File: KWOC/KWOC/ViewController.m
//
//  ViewController.m
//  KWOC
//
//  Created by 渴望 on 2023/6/14.
//

#import "ViewController.h"
#import "TabBarDemoVC.h"
#import <KW-Swift.h>
#import "SizeO.h"
#import "KWColorDemo.h"
#import "KWNavigationVC.h"
#import "TestBaseVC.h"
#import "TestTableView.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>
@property (nonatomic, copy) NSArray *dataArr;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = [UIColor redColor];
    
    self.dataArr = @[@"KWTabBar",
                     @"KWColor",
                     @"KWNavigationController",
                     @"KWViewController",
                     @"KWTableView"];
    
    UITableView * tableView = [[UITableView alloc]initWithFrame:CGRectMake(0, kNavigationStatusHeight(), kScreenWidth(), kScreenHeight()-kNavigationStatusHeight()) style:UITableViewStyleGrouped];
    tableView.delegate = self;
    tableView.dataSource = self;
    tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
    tableView.estimatedSectionFooterHeight = 0;
    tableView.estimatedSectionHeaderHeight = 0;
    if (@available(iOS 15.0, *)) {
        tableView.sectionHeaderTopPadding = 0;
    } else {
        // Fallback on earlier versions
    }
    [self.view addSubview:tableView];
    
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.dataArr.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell * cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:[NSString stringWithFormat:@"%ld%ld",(long)indexPath.section,(long)indexPath.row]];
    cell.textLabel.text = self.dataArr[indexPath.row];
    cell.backgroundColor = UIColor.whiteColor;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if (indexPath.row == 0) {
        TabBarDemoVC * vc = [[TabBarDemoVC alloc]init];
        [self.navigationController pushViewController:vc animated:true];
    }else if (indexPath.row == 1) {
        KWColorDemo * vc = [[KWColorDemo alloc]init];
        [self.navigationController pushViewController:vc animated:true];
    }else if (indexPath.row == 2) {
        KWNavigationVC * vc = [[KWNavigationVC alloc]init];
        KWNavigationController * nav = [[KWNavigationController alloc]initWithRootViewController:vc];
        [UIApplication sharedApplication].windows.firstObject.rootViewController = nav;
        [[UIApplication sharedApplication].windows.firstObject makeKeyAndVisible];
    }else if (indexPath.row == 3) {
        TestBaseVC * vc = [[TestBaseVC alloc]init];
        KWNavigationController * nav = [[KWNavigationController alloc]initWithRootViewController:vc];
        [UIApplication sharedApplication].windows.firstObject.rootViewController = nav;
        [[UIApplication sharedApplication].windows.firstObject makeKeyAndVisible];
    }else if (indexPath.row == 4) {
        TestTableView * vc = [[TestTableView alloc]init];
        KWNavigationController * nav = [[KWNavigationController alloc]initWithRootViewController:vc];
        [UIApplication sharedApplication].windows.firstObject.rootViewController = nav;
        [[UIApplication sharedApplication].windows.firstObject makeKeyAndVisible];
    }
}







@end

File: KW/KW/ViewController.swift
//
//  ViewController.swift
//  KW
//
//  Created by 渴望 on 2023/6/13.
//

import UIKit

class ViewController: UIViewController {

    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        view.backgroundColor = UIColor.red;
        
        view.addSubview(tableView)
    }

    let dataArr:[String] = ["KWTabBar",
    "KWColor",
    "KWNavigation",
    "KWViewController",
    "KWTableViewController",
    "KWCollectionViewController"];
    
    
    lazy var tableView: UITableView =  {
        let table = UITableView.init(frame: CGRect(x: 0, y: KNavigationStatusHeight, width: KScreenWidth, height: KScreenHeight-KNavigationStatusHeight), style: .plain)
        table.dataSource = self
        table.delegate = self
        table.separatorStyle = .none
        table.tableFooterView = UIView()
        table.tableHeaderView = UIView.init(frame: CGRect.init(origin: .zero, size: CGSize(width: 1, height: CGFloat.leastNormalMagnitude)))
        table.estimatedSectionHeaderHeight = 0
        table.estimatedSectionFooterHeight = 0
        if #available(iOS 11.0, *) {
            table.contentInsetAdjustmentBehavior = .never
        } else {
            self.automaticallyAdjustsScrollViewInsets = false
        }
        if #available(iOS 15.0, *) {
            table.sectionHeaderTopPadding = 0
        }
        return table
    }()
    
    let ss = CGRect(x: KAdapt(20), y: KAdapt(20), width: KAdapt(100), height: KAdapt(100))
}

extension ViewController: UITableViewDelegate, UITableViewDataSource {
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataArr.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = UITableViewCell(style: .default, reuseIdentifier: NSString(format: "%d%d",indexPath.section,indexPath.row) as String)
        
        cell.textLabel?.text = dataArr[indexPath.row]
        cell.backgroundColor = .KW_L(.section)
        return cell
    }
    
    // MARK: - UITableViewDelegate
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 45
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        if indexPath.row == 0 {
            let vc = TabBarDemoVC()
            self.navigationController?.pushViewController(vc, animated: true)
        }else if indexPath.row == 1 {
            let vc = KWColorDemoVC()
            self.navigationController?.pushViewController(vc, animated: true)
        }else if indexPath.row == 2 {
            let vc = KWNavigationVC()
            let nav = KWNavigationController(rootViewController: vc)
            self.changeRootViewController(vc: nav)
        }else if indexPath.row == 3 {
            let vc = TestBaseVC()
            let nav = KWNavigationController(rootViewController: vc)
            self.changeRootViewController(vc: nav)
        }else if indexPath.row == 4 {
            let vc = TestTableView()
            let nav = KWNavigationController(rootViewController: vc)
            self.changeRootViewController(vc: nav)
        }else if indexPath.row == 5 {
            let vc = TestCollectionVC()
            let nav = KWNavigationController(rootViewController: vc)
            self.changeRootViewController(vc: nav)
        }
        
    }
    
    
    
    
    
    
    
    
    
    
    
    
    func changeRootViewController(vc:UIViewController) {
        if #available(iOS 15.0, *) {
            let scenes = UIApplication.shared.connectedScenes
            let windowScene = scenes.first as? UIWindowScene

            if let windowScene = windowScene {
                windowScene.keyWindow?.rootViewController = vc
                windowScene.keyWindow?.makeKeyAndVisible()
            }
        }else{
            UIApplication.shared.windows.first?.rootViewController = vc
            UIApplication.shared.windows.first?.makeKeyAndVisible()
        }
    }
    
    
}

